src/lib/coda_state/global_state.ml

```diff
@@ -0,0 +1,56 @@
+open Core_kernel
+open Coda_base
+open Snark_params.Tick
+
+module Poly = struct
+  [%%versioned
+  module Stable = struct
+    module V1 = struct
+      type 'token_id t = {next_available_token: 'token_id}
+      [@@deriving sexp, eq, compare, hash, yojson]
+    end
+  end]
+
+  type 'token_id t = 'token_id Stable.Latest.t =
+    {next_available_token: 'token_id}
+  [@@deriving sexp, eq, compare, fields, yojson]
+
+  let to_hlist {next_available_token} = H_list.[next_available_token]
+
+  let of_hlist ([next_available_token] : (unit, _) H_list.t) =
+    {next_available_token}
+end
+
+type 'token_id poly = 'token_id Poly.t = {next_available_token: 'token_id}
+[@@deriving sexp, eq, compare, fields, hash, yojson]
+
+module Value = struct
+  [%%versioned
+  module Stable = struct
+    module V1 = struct
+      type t = Token_id.Stable.V1.t Poly.Stable.V1.t
+      [@@deriving sexp, eq, compare, hash, yojson]
+
+      let to_latest = Fn.id
+    end
+  end]
+
+  type t = Stable.Latest.t
```

Comment:
should this type have the same deriving clause as the versioned type?

Reply:
You're right. Updated.